core/lib/dal/src/eth_sender_dal.rs

```diff
@@ -23,10 +23,83 @@ pub struct EthSenderDal<'a, 'c> {
 }
 
 impl EthSenderDal<'_, '_> {
+    pub async fn get_non_final_txs(
+        &mut self,
+        operator_address: Address,
+        is_gateway: bool,
+    ) -> sqlx::Result<Vec<EthTx>> {
+        let txs = sqlx::query_as!(
+            StorageEthTx,
+            r#"
+            SELECT
+                eth_txs.*
+            FROM
+                eth_txs
+            JOIN eth_txs_history ON eth_txs.confirmed_eth_tx_history_id = eth_txs_history.id
+            WHERE
+                from_addr = $1
+                AND is_gateway = $2
+                AND eth_txs_history.finality_status != 'finalized'
+            "#,
+            operator_address.as_bytes(),
+            is_gateway,
+        )
+        .fetch_all(self.storage.conn())
+        .await?;
+        Ok(txs.into_iter().map(|tx| tx.into()).collect())
+    }
+
+    pub async fn unfinalize_txs(
+        &mut self,
+        operator_address: Address,
+        is_gateway: bool,
+        from_eth_tx_id: u32,
```

Comment:
will this work when we have a different operator address for COMMITs? Or do you only use it for GW? Still, I'd prefer to iterate over concrete transactions that need to be unfinalized explicitely

Reply:
Everything is the same as before. We treat txs based on operator address